framework/src/st_things/things_stack/easy-setup/easysetup_manager.c

```diff
@@ -235,34 +235,16 @@ esm_result_e esm_init_easysetup(int restart_flag, things_server_builder_s *serve
 		THINGS_LOG_D(THINGS_INFO, TAG, "Restart EasySetup");
 		esm_continue = 0;
 		if (gthread_id_cloud_refresh_check) {
-#ifdef __ST_THINGS_RTOS__
-			if (ci_token_expire_fds[1] != -1) {
-				ssize_t len = 0;
-				do {
-					len = write(ci_token_expire_fds[1], "w", 1);
-				} while ((len == -1) && (errno == EINTR));
-				if ((len == -1) && (errno != EINTR) && (errno != EPIPE)) {
-					THINGS_LOG_D_ERROR(THINGS_ERROR, TAG, "write failed: %s", strerror(errno));
-					return ESM_ERROR;
-				}
-			}
-			pthread_join(gthread_id_cloud_refresh_check, NULL);
-			if (ci_token_expire_fds[0] != -1) {
-				close(ci_token_expire_fds[0]);
-				ci_token_expire_fds[0] = -1;
-			}
-			if (ci_token_expire_fds[1] != -1) {
-				close(ci_token_expire_fds[1]);
-				ci_token_expire_fds[1] = -1;
-			}
-#else
+ 			pthread_cancel(gthread_id_cloud_refresh_check);
 			pthread_join(gthread_id_cloud_refresh_check, NULL);
-#endif
+			pthread_detach(gthread_id_cloud_refresh_check);
```

Comment:
Don't use both, pthread_join and pthread_detach. You shoule use only one function among them.

Reply:
Ok. remove pthread_detach